lib/Model/SettingsMapper.php

```diff
@@ -32,16 +37,29 @@ public function __construct(IDBConnection $db) {
 	 * @return Settings
 	 * @throws DBException
 	 * @throws MultipleObjectsReturnedException
-	 * @throws DoesNotExistException
 	 */
 	public function getSettingsByUser(string $userId): Settings {
-		$query = $this->db->getQueryBuilder();
+		try {
+			$query = $this->db->getQueryBuilder();
 
-		$query->select('*')
-			->from($this->getTableName())
-			->where($query->expr()->eq('user_id', $query->createNamedParameter($userId)));
+			$query->select('*')
+				->from($this->getTableName())
+				->where($query->expr()->eq('user_id', $query->createNamedParameter($userId)));
 
-		return $this->findEntity($query);
+			return $this->findEntity($query);
+		} catch (DoesNotExistException) {
+			$defaultBatchtime = (int)$this->appConfig->getValueString(Application::APP_ID, 'setting_batchtime');
+
+			if ($defaultBatchtime !== Settings::EMAIL_SEND_WEEKLY
+				&& $defaultBatchtime !== Settings::EMAIL_SEND_DAILY
+				&& $defaultBatchtime !== Settings::EMAIL_SEND_3HOURLY
+				&& $defaultBatchtime !== Settings::EMAIL_SEND_HOURLY
+				&& $defaultBatchtime !== Settings::EMAIL_SEND_OFF) {
+				$defaultBatchtime = Settings::EMAIL_SEND_3HOURLY;
+			}
+
+			return $this->setBatchSettingForUser($userId, $defaultBatchtime);
```

Comment:
this runs the `findEntity` again, so it's duplicate, but the code flow is "clear"